lib/data_sources/directory_locator.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require "utils/file_transfer"
+
+module DataSources
+  # Given a base and and org id, provide paths (strings) to the org's dirs.
+  # Assumes identical structure on the remote and local side,
+  # just with a different base path, and does not know/care if the paths
+  # that it provides actually point to anything real.
+  # E.g.:
+  # remote_d = DataSources::DirectoryLocator.new("dropbox:foo/bar/member_data", "foo")
+  # local_d = DataSources::DirectoryLocator.new("/tmp/member_data", "foo")
+  # remote_holdings = remote_d.holdings_current
+  # local_holdings = local_d.holdings_current
+  class DirectoryLocator
+    attr_reader :root, :organization
+    def initialize(root, organization)
+      @root = root # the part of the file system where the member directories start
+      @organization = organization
+      # In the off chance the object is created 1s before jan 1st,
+      # at least we'll be consistent across the life of this object.
+      @year = Time.new.year.to_s
+    end
+
+    # The base directory for the organization.
+    def base
+      join(@root, "#{organization}-hathitrust-member-data")
```

Comment:
I'd suggest removing the private method and using `File.join()` instead

Reply:
ok